src/languages/swift.js

```diff
@@ -98,13 +111,72 @@ export default function(hljs) {
   };
   SUBST.contains = [NUMBER];
 
+  // https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID418
+  const operatorHead = either(
+    /[/=\-+!*%<>&|^~?]/,
+    /[\u00A1–\u00A7]/,
+    /[\u00A9\u00AB]/,
+    /[\u00AC\u00AE]/,
+    /[\u00B0\u00B1]/,
+    /[\u00B6\u00BB\u00BF\u00D7\u00F7]/,
+    /[\u2016–\u2017]/,
+    /[\u2020–\u2027]/,
+    /[\u2030–\u203E]/,
+    /[\u2041–\u2053]/,
+    /[\u2055–\u205E]/,
+    /[\u2190–\u23FF]/,
+    /[\u2500–\u2775]/,
+    /[\u2794–\u2BFF]/,
+    /[\u2E00–\u2E7F]/,
+    /[\u3001–\u3003]/,
+    /[\u3008–\u3020]/,
+    /[\u3030]/
+  );
+  const operatorCharacter = either(
+    operatorHead,
+    /[\u0300–\u036F]/,
+    /[\u1DC0–\u1DFF]/,
+    /[\u20D0–\u20FF]/,
+    /[\uFE00–\uFE0F]/,
+    /[\uFE20–\uFE2F]/,
+    // TODO: The following characters are also allowed, but the regex doesn't work as intended.
+    // For example, it also matches -10 as an operator.
+    // /[\u{E0100}–\u{E01EF}]/u
```

Comment:
This doesn't seem right.  From what I'm reading this is the full range of what MIGHT be used for defining a custom operator... just because something _could potentially_ be an operator doesn't mean it most assuredly IS an operator and should be highlighted everywhere as an operator... correct?

This "wideness" might make sense in the context of something like `func [x] (a,b)` where [x] is an operator being defined (and we know that by context), but I don't think it makes sense in general.  Pretty sure in the standard case we should only highlight well-known built-in operators that Swift supports, like `>`, `<`, etc...

Reply:
Some of these characters are used elsewhere in the grammar, such as `<GenericParam>`, or `OptionalType?`, but other than that, they're always operators.

I assume the set of "head" character for operators and other identifiers is disjunct. For example, if you look at the range A1-FF, the ones that are missing from the list here are listed explicitly as valid head characters for regular (non-operator) identifiers:

\u00A8\u00AA\u00AD\u00AF\u00B2–\u00B5\u00B7–\u00BA
\u00BC–\u00BE\u00C0–\u00D6\u00D8–\u00F6\u00F8–\u00FF

So I think this should work just fine, as long as the cases where operators are used as punctuation are handled by specific modes. In the case of generics and optional types, that's already the case.